include/zephyr/drivers/opamp.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_OPAMP_H_
+#define ZEPHYR_INCLUDE_DRIVERS_OPAMP_H_
+
+/**
+ * @brief Opamp Interface
+ * @defgroup opamp_interface Opamp Interface
+ * @since 4.3
+ * @version 0.1.0
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/device.h>
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief OPAMP gain factors. */
+enum opamp_gain {
```

Comment:
Maybe something like would be much better, since no direct gain values will be given in .h file
```suggestion
typedef size_t opamp_gain;
```